signal_scanner_bot/transport.py

```diff
@@ -1,72 +1,108 @@
 """Main module."""
+import pprint
 import asyncio
 import logging
 import subprocess
 import concurrent.futures
-import urllib3
 
-import tweepy
+import peony
 import ujson
 
 from . import env
 from . import messages
-from . import twitter
 from . import signal
 
 log = logging.getLogger(__name__)
 
 
 ################################################################################
-# Stream Listener
+# Twitter-to-Queue
 ################################################################################
-class Listener(tweepy.StreamListener):
-    """
-    Listener implementation for when a status matching the track is received.
-    """
-
-    def on_status(self, status: tweepy.Status):
-        if not env.STATE.LISTENING:
-            return
-
-        messages.process_twitter_message(status)
+def _filter_hashtags(data, filter_hashtag_list):
+    for input_hashtag in data["entities"]["hashtags"]:
+        if input_hashtag["text"] in filter_hashtag_list:
+            return True
+    return False
 
 
-################################################################################
-# Twitter-to-Queue
-################################################################################
-def _twitter_to_queue():
+class Stream(peony.PeonyClient):
     """
-    Top level function for running the twitter-to-signal loop.
+    Base class for the Event Stream
+    Nothing is changed here as we are not editing the base class
     """
-    api = twitter.get_api()
-    stream = tweepy.Stream(auth=api.auth, listener=Listener())
-    log.info("Twitter stream initialized, starting to follow")
-    stream.filter(track=env.RECEIVE_HASHTAGS, stall_warnings=True)
+
+    pass
+
+
+@Stream.event_stream
+class UserStream(peony.EventStream):
+    async def stream_request(self):
+        """
+        Basic stream set up
+        """
+        await self.api.client.stream.statuses.filter.post(
+            follow=",".join(env.TRUSTED_TWEETERS)
+        )
+
+    @peony.events.on_connected.handler
+    def connection(self):
+        """
+        Action to perform on connection
+        """
+        print("Connected to Twitter Stream API")
+
+    @peony.events.on_retweeted_status.handler
+    async def on_retweet(self, data):
+        """
+        Filter tweets from followed users on specified hashtags
+        Execute something (send to Signal or whatever)
+        """
+        pass
+
+    @peony.events.on_tweet.handler
+    async def on_tweet(self, data):
+        """
+        Filter tweets from followed users on specified hashtags
+        Execute something (send to Signal or whatever)
+        """
+        print("Got a tweet\n\n\n\n\n")
+        if _filter_hashtags(data, env.RECEIVE_HASHTAGS):
+            pprint.pprint(data)
```

Comment:
The handler for when a tweet event happens in the event stream. Right now obviously it's just checking for the tweet then printing it out. I tested this in a separate script, but can't get this to trigger when it's run within the coroutine loops.